docs/docs/3-features-modules/10-warning-module/index.md

```diff
@@ -0,0 +1,278 @@
+@Entity()
+class NewsPageType {
+    @Field({ nullable: true })
+    @Property({ columnType: "text", nullable: true })
+    newsId?: string;
+}
+```
+
+#### Example of the Service:
+
+```typescript
+class NewsWarningService implements CreateWarningsServiceInterface<NewsPageType> {
+    constructor(
+        @InjectRepository(News)
+        private readonly newsRepository: EntityRepository<News>,
+    ) {}
+
+    async createWarnings(entity: MyEntity): Promise<EntityWarning[]> {
+        const warnings: EntityWarning[] = [];
+
+        if (entity.newsId) {
+            const news = await this.newsRepository.findOne(entity.newsId);
+            if (!news) {
+                warnings.push({ severity: "high", message: "invalidNewsTarget" });
+            }
+        }
+
+        return warnings;
+    }
+}
+```
+
+### Entity Warning for improved performance
+
+When defining a warning with one of the above methods, performance issues may arise if your entity's database table contains numerous entries. In such cases, the `WarningService` provides a generator function `async *bulkCreateWarnings()` that allows for more efficient querying and checking of warnings.
+
+Instead of checking all records (which happens with the regular `createWarnings` function), you can define specific criteria for which entities should be checked. For example, if you only need to check warnings for active `News` items (excluding archived ones), you can reduce the number of records to check.
+
+Here's how to implement this:
+
+```typescript
+class NewsWarningService implements CreateWarningsServiceInterface {
+    constructor(
+        @InjectRepository(News)
+        private readonly newsRepository: EntityRepository<News>,
+    ) {}
+
+    async *bulkCreateWarnings() {
+        let offset = 0;
+        const limit = 50;
+
+        let news = [];
+        do {
+            news = await this.newsRepository.find(
+                {
+                    isArchived: false,
+                },
+                { limit, offset },
+            );
+
+            for (const entity of news) {
+                if (!entity.title) {
+                    yield {
+                        warnings: [
+                            {
+                                message: "missingTitle",
+                                severity: "low",
+                            },
+                        ],
+                        targetId: news.id,
+                        scope: news.scope,
+                    };
+                }
+            }
+            offset += limit;
+        } while (news.length > 0);
+    }
+}
+```
+
+## Admin: Custom Block Warning Messages
+
+In the API, you defined custom warning messages for your blocks or entities. To make these messages translatable in the Admin, you need to define them in your `App.tsx` using the `CometConfigProvider`:
+
+```tsx
+<CometConfigProvider
+    {...otherConfigs}
+    warnings: {
+        warningMessages: {
```

Comment:
```suggestion
        messages: {
```

Reply:
true I changed that yesterday in the other PR: 8bca19f78717b8be8a3a759d648c799960231c7a